pkg/client/clientcmd/validation.go

```diff
@@ -109,7 +109,8 @@ func validateClusterInfo(clusterName string, clusterInfo clientcmdapi.Cluster) [
 	if len(clusterInfo.Server) == 0 {
 		validationErrors = append(validationErrors, fmt.Errorf("no server found for %v", clusterName))
 	}
-	if len(clusterInfo.CertificateAuthority) != 0 {
+	// Only validate the CA file if it hasn't been overridden by CA data
```

Comment:
looks like validation only runs when it's turning a .kubeconfig into a client.Config, not when using `kubectl config` commands to update the config. I updated the validation and updated the `kubectl` commands to clear the *data fields when setting the file fields.